catalog/ui/search-ui/search-endpoint/src/main/java/org/codice/ddf/ui/searchui/query/controller/search/QueryRunnable.java

```diff
@@ -289,6 +286,31 @@ protected QueryResponse queryCatalog(String sourceId, SearchRequest searchReques
         return response;
     }
 
+    private void collectResponses(QueryRequest request, QueryResponse response)
+            throws UnsupportedQueryException, SourceUnavailableException, FederationException {
+        int intialResultsSize = response.getResults()
+                .size();
+        while (intialResultsSize < response.getHits() && intialResultsSize < request.getQuery()
```

Comment:
...sigh, this means there is no way I can prevent an extra query if the results returned are less than the query page size